staging/src/k8s.io/apiserver/plugin/pkg/authenticator/token/oidc/oidc.go

```diff
@@ -221,6 +221,10 @@ var allowedSigningAlgs = map[string]bool{
 	oidc.PS512: true,
 }
 
+// BaseContext is the root context used when spawning go routines.
+// Exported so that it can be overridden in integration tests.
+var BaseContext = context.Background()
```

Comment:
We do cancel the go routines for the old ones, but only after we swap in something to replace them with.  This `BaseContext` is necessary right now for the case of "I swapped in a structurally valid config that has a typo in the issuer URL."  Since the final config is invalid, we will keep trying forever to set up the authenticator.

Reply:
Fixed.